src/qibolab/platforms/platform.py

```diff
@@ -32,16 +120,50 @@ def disconnect(self):
         self.design.disconnect()
         self.is_connected = False
 
-    def execute_pulse_sequence(self, sequence, nshots=1024, relaxation_time=None, raw_adc=False):
-        if relaxation_time is None:
-            relaxation_time = self.relaxation_time
-        return self.design.play(self.qubits, sequence, nshots=nshots, relaxation_time=relaxation_time, raw_adc=raw_adc)
+    def execute_pulse_sequence(self, sequence, options, **kwargs):
+        """Executes a pulse sequence.
+
+        Args:
+            sequence (:class:`qibolab.pulses.PulseSequence`): Pulse sequence to execute.
+            options (:class:`qibolab.platforms.platform.ExecutionParameters`) Class holding the execution options.
+            **kwargs: May need them for something
+        Returns:
+            Readout results acquired by after execution.
+        """
+        if options.nshots is None:
+            options = replace(options, nshots=self.settings["settings"]["nshots"])
```

Comment:
Since we are "refactoring" I think that we could add a `nshots` attribute in the `AbstractPlatform`. I would like to remove the `settings` dictionary as soon as possible.

Reply:
done